packages/adapters/bitcoin/src/connectors/UnisatConnector.ts

```diff
@@ -0,0 +1,265 @@
+
+    let txid: string | undefined = undefined
+    if (params.broadcast) {
+      txid = await this.wallet.pushPsbt(signedPsbtHex)
+    }
+
+    return {
+      psbt: Buffer.from(signedPsbtHex, 'hex').toString('base64'),
+      txid
+    }
+  }
+
+  public async switchNetwork(caipNetworkId: string): Promise<void> {
+    const network = this.getNetwork(caipNetworkId)
+
+    await this.wallet.switchChain(network)
+  }
+
+  public request<T>(_args: RequestArguments): Promise<T> {
+    return Promise.reject(new MethodNotSupportedError(this.id, 'request'))
+  }
+
+  private bindEvents(): void {
+    this.unbindEvents()
+
+    this.wallet.on('accountsChanged', account => {
+      if (typeof account === 'object' && account && 'address' in account) {
+        this.emit('accountsChanged', [account.address])
+      }
+    })
+  }
+
+  private unbindEvents(): void {
+    this.wallet.removeListener('accountsChanged', account => {
+      if (typeof account === 'object' && account && 'address' in account) {
+        this.emit('accountsChanged', [account.address])
+      }
+    })
+  }
+
+  public static getWallet(params: UnisatConnector.GetWalletParams): UnisatConnector | undefined {
+    if (!CoreHelperUtil.isClient()) {
+      return undefined
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any, init-declarations
+    let wallet: any
+
+    switch (params.id) {
+      case 'unisat':
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        wallet = (window as any)?.unisat
+        break
+      case 'bitget':
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        wallet = (window as any)?.bitkeep?.unisat
+        break
+      case 'binancew3w':
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        wallet = (window as any)?.binancew3w?.bitcoin
+        break
+      default:
+        throw new Error(`Unsupported wallet id: ${params.id}`)
+    }
+
+    if (wallet) {
+      return new UnisatConnector({ wallet, imageUrl: '', ...params })
+    }
+
+    return undefined
+  }
+
+  public async getPublicKey(): Promise<string> {
+    return this.wallet.getPublicKey()
+  }
+
+  private getNetwork(caipNetwork: string): UnisatConnector.Chain {
+    switch (caipNetwork) {
+      case bitcoin.caipNetworkId:
+        return 'BITCOIN_MAINNET'
+      case bitcoinTestnet.caipNetworkId:
+        return 'BITCOIN_TESTNET'
+      default:
+        throw new Error('UnisatConnector: unsupported network')
+    }
+  }
+}
+
+export namespace UnisatConnector {
+  export type Chain = 'BITCOIN_MAINNET' | 'BITCOIN_TESTNET' | 'FRACTAL_BITCOIN_MAINNET'
+  export type Network = 'livenet' | 'testnet'
+  export type Id = 'unisat' | 'bitget' | 'binancew3w'
+
+  export type ConstructorParams = {
+    id: Id
+    name: string
+    wallet: Wallet
+    requestedChains: CaipNetwork[]
+    getActiveNetwork: () => CaipNetwork | undefined
+    imageUrl: string
+  }
+
+  export type Wallet = {
```

Comment:
nit: may be worth to move these types to a different file. eg create a directory for UnisatConnector and do `index.ts` + `types.ts`

Reply:
What will happen to the namepsace on the index.ts?